File: MathAlgorithms.py
print("hello! these are some simple math algorithms for practice.")

print("address: https://github.com/amrezalou/simple-MathAlgorithms.git")

print("""
1) fibonacci series
2) show prime number
3) LCM
4) calculate factorial
5) check-odd-even
6) check perfect number
7) pythagorean numbers
8) discount calculation
9) area of some shapes
10) GCD
""")



# 1

def FibonacciSeries(n):
   """Function that calculates fibonacci series up to n."""
   result = []
   a, b = 0, 1
   while a < n:
       result.append(a)
       a, b = b, a+b
   print(result)


# -------------------------------------------------------------------------

# 2

def PrimeNumber(n):
    """ Shows prime numbers. """

    for num in range(2, n+1):
        is_prime = True
        for i in range(2, int(num ** 0.5) + 1):
            if num % i == 0:
                is_prime = False
                break
        if is_prime:
            print(num, "is prime.")
        else:
            print(num, "is not prime.")


#-------------------------------------------------------------------------

# 3

def LCM(a, b):
   """ Shows LCM between 2 numbers. """
   MAX = max(a, b)
   MIN = min(a, b)

   if MAX % MIN == 0:
       print("LCM (a, b):" , MAX)
   else:
       while(True):
           if MAX % a == 0 and MAX % b == 0:
               lcm = MAX
               break
           MAX+=1
       print("LCM (a, b): ", lcm)




#-------------------------------------------------------------------------

# 4

def Factorial(n):
   """ Calculate the factorial of n. """
   if n == 0 or n == 1:
       return 1
   else:
       return n * Factorial(n - 1)




#-------------------------------------------------------------------------

# 5


def ODD_and_EVEN(number):
   """ Check for type of number among odd and even. """

   if number % 2 == 0:
       print("the number is even!")
   else:
       print(number, "is odd!")



# -------------------------------------------------------------------------

# 6

def Perfect(n):
   result = []
   for i in range(1, n):
       if n % i == 0:
           result.append(i)
   sum_of_result = sum(result)
   return sum_of_result == n



#-------------------------------------------------------------------------

# 7

def pythagorean_number_checker():
   """ Check if 3 numbers are Pythagorean or they're not. """
   # a**2 + b**2 = c**2 (chord)
   a = int(input("enter the smallest number: "))
   b = int(input("enter the number with average value(second): "))
   c = int(input("enter the bigges one now: "))

   if a**2 + b**2 == c**2:
       print("True, these numbers are pythagorean numbers.")

   else:
       print("False, numbers are not pythagorean numbers.")




#-------------------------------------------------------------------------

# 8

def discount_calculation(price, precentageD):
   """ A simple fucntion to calculte discount of something product. """
   discount = price * precentageD / 100
   print(price - discount)




#-------------------------------------------------------------------------

# 9

import math

def Acircle(r):
   """ Computing area of circle. """

   aoc = r*r*math.pi
   print(aoc)


def Atriangle(b, h):
   """ Computing area of triangle. """

   aot = b*h*0.5
   print(aot)


def Atrapeziod(b1, b2, h):
   """ Computing area of trapeziod. """

   aot2 = ((b1+b2) / 2 ) * h
   print(aot2)



#-------------------------------------------------------------------------

# 10

def GCD(a, b):
   """ Compute GCD of two numbers."""

   small = min(a, b)

   for i in range (1, small + 1):
       if a % i == 0  and b % i == 0:
           gcd = i
   print(gcd)





user = input("which one do you want to see/use? (1 to 10): ")

if user == '1':
    n = int(input("enter the number of fibonacci up to the: "))
    FibonacciSeries(n)

elif user == '2':
    n = int(input("check range of prime numbers up to n: "))
    PrimeNumber(n)

elif user == '3':
    n1 = int(input("enter the first number: "))
    n2 = int(input("enter the first number: "))

    LCM(n1, n2)
elif user == '4':
    n = int(input("factorial of your number: "))
    print("factorial of", n, "is: ", Factorial(n))

elif user == '5':
    n = int(input("enter number for typecheck: "))
    ODD_and_EVEN(n)

elif user == '6':
    n = int(input("check if its perfect number: "))
    if Perfect(n):
        print(n, "is a perfect.")
    else:
        print(n, "is not a perfect.")

elif user == '7':
    pythagorean_number_checker()

elif user == '8':
    price = float(input("enter price: "))
    precentage = float(input("enter precentage: "))
    discount_calculation(price, precentage)

elif user == '9':
    ask = input("which shape? \n 1)circle \n 2)triangle \n 3)trapeziod: ")
    if ask == '1':
        n = float(input("enter value of radius of the circle: "))
        Acircle(n)

    elif ask == '2':
        b = float(input("enter base value: "))
        h = float(input("enter height value: "))
        Atriangle(b, h)

    else:
        b1 = float(input("base1 value: "))
        b2 = float(input("base2 value: "))
        h = float(input("and finally enter the value of height: "))
        Atrapeziod(b1, b2, h)

elif user == '10':
    n = int(input("the first: "))
    m = int(input("the second: "))
    GCD(n, m)

File: MathAlgorithms.cpp
#include <iostream>
using namespace std;

int main () {

    // list of options:

    cout << "1) fibonnaci series\n2) show prime number\n3) LCM\n4) calculate factorial\n\
5) ceck-odd-even\n6) check perfect number\n7) pythagorean numbers\n8) discount calculation\n\
9) GCD" << endl;

    // programs progress

    // 1:

/*    int n, t1 = 0, t2 = 1, nextTerm = 0;

    cout << "Enter the number of terms: ";
    cin >> n;

    cout << "Fibonacci Series: ";

    for (int i = 1; i <= n; ++i) {

        if(i == 1) {
            cout << t1 << " ";
            continue;
        }
        if(i == 2) {
            cout << t2 << " ";
            continue;
        }
        nextTerm = t1 + t2;
        t1 = t2;
        t2 = nextTerm;

        cout << nextTerm << " ";
    }
*/


    // 2:

 /*   int n;
    bool isPrime = true;

    cout << "Enter the number: ";
    cin >> n;

    for (int i = 2; i < n/2; i++) {
        if (n % i == 0) {
            isPrime = false;
            break;
        }
    }

    if (isPrime && n > 1) {
        cout << n << " is a prime number." << endl;
    }
    else {
        cout << n << " is not a prime number." << endl;
    }

*/


    // 3:

/*    int num1, num2;
    cout << "Enter the first number: ";
    cin >> num1;

    cout << "Enter the second number: ";
    cin >> num2;

    int max = (num1 > num2) ? num1 : num2;
    int lcm = max;

    while (true) {
        if (lcm % num1 == 0 && lcm % num2 == 0) {
            cout << "The LCM between " << num1 << " and " << num2 << " is: " << lcm << endl;
            break;
        }
        lcm += max;
    }

*/



    // 4:

/*    int number;
    int fa = 1;

    cout << "enter the number: ";
    cin >> number;

    for (int i = 1; i <= number; i++) {
        fa*=i;
    }
    cout << "factorial of " << number << " is: " << fa << endl;
*/


    // 5:

/*    int num;

    cout << "enter the number: ";
    cin >> num;

    if (num % 2 == 0)
        cout << num << " is a even number." << endl;
    else
        cout << num << " is a odd number." << endl;

*/



    // 6:

/*    int number;

    cout << "enter the number: ";
    cin >> number;

    int sum = 0;

    for (int i = 1; i <= number / 2; i++) {
        if (number % i == 0) {
            sum += i;
        }
    }
*/



    // 7:

/*    int num1, num2, num3;
    bool checker;

    cout << "enter smallest number: ";
    cin >> num1;

    cout << "enter the second number: ";
    cin >> num2;

    cout << "enter the biggest number: ";
    cin >> num3;

    if ( (num1*num1) + (num2*num2) == (num3*num3) ) {
        checker = true;
    }
    else
        checker = false;

    if (checker == true)
        cout << "pythagorean numbers" << endl;
    else
        cout << "not pythagorean numbers" << endl;

*/


    // 8:

/*    float discount, price, precentageD;

    cout << "enter price of product: ";
    cin >> price;

    cout << "enter precent of discount: ";
    cin >> precentageD;

    discount = (price * precentageD) / 100;

    cout << "final price: " << price-discount << endl;

*/


    // 9:

 /*  int n1, n2, gcd;

    cout << "first number: ";
    cin >> n1;

    cout << "second number: ";
    cin >> n2;

    int min = (n1 < n2) ? n1 : n2;

    for (int i = 1; i < min + 1; i++) {
        if (n1 % i == 0 && n2 % i == 0) {
            gcd = i;
        }
    }

    cout << "gcd of " << n1 << " and " << n2 << " is: " << gcd << endl;
*/

    return 0; // program finished successfully.

}
